fish-rust/src/builtins/shared.rs

```diff
@@ -61,6 +63,11 @@ impl output_stream_t {
     pub fn append<Str: AsRef<wstr>>(&mut self, s: Str) -> bool {
         self.ffi().append1(c_str!(s))
     }
+
+    /// Append a char.
+    pub fn append1(&mut self, c: char) -> bool {
+        self.append(wstr::from_char_slice(&[c]))
+    }
 }
```

Comment:
Just fyi, the rust convention is `push_str()` and `push()` so you could use `append_str()` and `append()` if you like. Or just `append()` and `push()` for now if the goal is to minimize differences with the cpp code.